modules/product/config/optional/core.entity_form_display.commerce_line_item.product_variation.add_to_cart.yml

```diff
@@ -0,0 +1,20 @@
+langcode: en
+status: true
+dependencies:
+  config:
+    - commerce_order.commerce_line_item_type.product_variation
+    - core.entity_form_mode.commerce_line_item.add_to_cart
+  module:
+    - commerce_product
+id: commerce_line_item.product_variation.add_to_cart
+targetEntityType: commerce_line_item
+bundle: product_variation
+mode: add_to_cart
```

Comment:
Note how this form display references the "add_to_cart" form mode, but that form mode is not exported.
Config dependencies are static, you can't rely on the result of hook_entity_bundle_create() being available. That's probably the cause of one of your crashes.